assets/js/translator.js

```diff
@@ -350,13 +339,391 @@ if(modeEnabled('translation')) {
+                        if(dists[mini] === undefined) {
+                            dists[mini] = Number.MAX_SAFE_INTEGER;
+                            newdist = Number.MAX_SAFE_INTEGER;
+                        }
+                        else {
+                            newdist = dists[mini] + weights[mini][trgt];
+                        }
+                        if((dists[trgt] === undefined) || (dists[mini] + weights[mini][trgt] < dists[trgt])) {
+                            dists[trgt] = newdist;
+                            prevs[trgt] = mini;
+                        }
+                    }
+                });
+            }
+        }
+
+        paths[src] = {};
+        $.each(Object.keys(prevs), function (i, trgt) {
+            var prev = trgt;
+            var path = [trgt];
+            while(prevs[prev]) {
+                prev = prevs[prev];
+                path.push(prev);
+            }
+            path.reverse();
+            paths[src][trgt] = {'path': path, 'weight': getWeight(path)};
+        });
+    });
+
+    $.each(srcLangs, function (i, src) {
+        chainedPairs[src] = Object.keys(paths[src]);
+    });
+
+    return paths;
+    // chainedPaths[src][dst] = [{path: [a, b, c], weight: 2}, {path: [a, d, e, c], weight: 3}]
+}
+
+function initChainGraph() {
+    var choose = d3.select('#chooseModalBody');
+    svg = choose
+        .append('svg')
+        .attr('width', width.toString() + 'px')
+        .attr('height', height.toString() + 'px');
+    choose.append('br');
+    choose.append('div').attr('id', 'validPaths')
+        .append('b')
+        .text('Valid Paths:')
+        .append('br');
+
+    /* eslint-disable no-magic-numbers */
+    svg.append('svg:defs').append('svg:marker')
+        .attr('id', 'end-arrow')
+        .attr('viewBox', '0 -5 10 10')
+        .attr('refX', 6)
+        .attr('markerWidth', 3)
+        .attr('markerHeight', 3)
+        .attr('orient', 'auto')
+        .append('svg:path')
+        .attr('d', 'M0,-5L10,0L0,5')
+        .attr('fill', '#999');
+
+    svg.append('svg:defs').append('svg:marker')
+        .attr('id', 'start-arrow')
+        .attr('viewBox', '0 -5 10 10')
+        .attr('refX', 4)
+        .attr('markerWidth', 3)
+        .attr('markerHeight', 3)
+        .attr('orient', 'auto')
+        .append('svg:path')
+        .attr('d', 'M10,-5L0,0L10,5')
+        .attr('fill', '#999');
+
+    simulation = d3.forceSimulation()
+        .force('link', d3.forceLink().id(function (d) { return d.id; }).distance(nodeSize * 10))
+        .force('charge', d3.forceManyBody().strength(-700))
+        .force('center', d3.forceCenter(width / 2, height / 2))
+        .alphaDecay(0.018);
+    /* eslint-enable no-magic-numbers */
+
+    refreshChainGraph();
+}
+
+function boundary(dist, max) {
+    if(dist < nodeSize) return nodeSize;
+    if(dist < max - nodeSize) return dist;
+    return max - nodeSize;
+}
+
+function getAllPaths(src, trgt, curPath, seens) {
```

Comment:
Similar short comment inside this function would be appreciated. Also, we're using both `trgt` and `dst` synonymously. We should really stick to one of them and probably the latter because it would be buggier to switch all the `dst` to `trgt` even though in retrospect I like target better.

Reply:
Should I change all the `trgt` to `dst` then, even inside other variable names? I will make it a separate commit so that the more important changes don't get swamped out.